rotkehlchen/tasks/calendar.py

```diff
@@ -47,3 +70,98 @@ def delete_past_calendar_entries(database: DBHandler) -> None:
+
+    ens_to_event: dict[str, tuple[int, EvmEvent]] = {}
+    for ens_event in registrations + renewals:
+        if (
+            ens_event.location_label is not None and
+            ens_event.extra_data is not None and
+            'fullname' in ens_event.extra_data and
+            'expires' in ens_event.extra_data and
+            ens_event.extra_data['expires'] > ts_now()  # those with expiry in the future
+        ):
+            ens_name: str = ens_event.extra_data['fullname']
+            ens_expires: Timestamp = ens_event.extra_data['expires']
+            blockchain = ChainID.deserialize(ens_event.location.to_chain_id()).to_blockchain()
+            if (calendar_entries := db_calendar.query_calendar_entry(
+                filter_query=CalendarFilterQuery.make(
+                    and_op=True,
+                    name=ens_name,
+                    addresses=[OptionalBlockchainAddress(
+                        blockchain=blockchain,
+                        address=string_to_evm_address(ens_event.location_label),
+                    )],
+                    blockchain=blockchain,
+                    counterparty=CPT_ENS,
+                ),
+            ))['entries_found'] == 0:  # if calendar entry doesn't exist, add it
+                if ens_event.location_label not in blockchain_accounts.get(blockchain):
+                    continue  # if the account doesn't exist, skip it
+
+                calendar_entry_id = db_calendar.create_calendar_entry(CalendarEntry(
+                    name=f'{ens_name} expiry',
+                    timestamp=ens_expires,
+                    description=f'{ens_name} expires on {customizable_date.timestamp_to_date(ens_expires)}',  # noqa: E501
+                    counterparty=CPT_ENS,
+                    address=string_to_evm_address(ens_event.location_label),
+                    blockchain=ChainID.deserialize(ens_event.location.to_chain_id()).to_blockchain(),
+                    color=None,
+                    auto_delete=True,
+                ))
+            else:  # else
+                calendar_entry_id = calendar_entries['entries'][0].identifier
+
+            if (
+                ens_name not in ens_to_event or
+                ens_expires > ens_to_event[ens_name][1].extra_data['expires']  # type: ignore[index]  # extra_data is not None, checked above
+            ):  # insert mapping for the latest expiry timestamp
+                ens_to_event[ens_name] = (calendar_entry_id, ens_event)
+
+    _, failed_to_add = db_calendar.create_reminder_enties(reminders=[
+        ReminderEntry(
+            identifier=idx,  # this is only used for logging below, it's auto generated in db
+            event_id=calendar_identifier,
+            secs_before=secs_before,
+        )
+        for idx, (calendar_identifier, _) in enumerate(ens_to_event.values())
+        for secs_before in (WEEK_IN_SECONDS, DAY_IN_SECONDS)
+    ])
```

Comment:
mmm checking it the `calendar_reminders` doesn't have any kind of check so you need to avoid creating reminders for calendar entries that already have them